### PR TITLE
Use github actions

### PR DESCRIPTION
**Pull request by stevemessick**
This runs unit tests on Windows! And does everything else that Travis does.

We can ignore the Kokoro failure. That problem is already fixed but I don't want to merge it until later. Travis hasn't even started yet so I cancelled it.

I still need to update the plugin tool to auto-generate the versions we build. I'll probably drop Travis in that PR.

### CODE REVIEW

**Comment by stevemessick** [comment]
I see I need to do some clean up in the Windows script. I'll get that in the next PR with the plugin tool changes.

File: tool/github.sh [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+export JAVA_HOME=$JAVA_HOME_11_X64
+
+git clone --depth 1 https://github.com/flutter/flutter.git ../flutter
+export PATH="$PATH":`pwd`/../flutter/bin:`pwd`/../flutter/bin/cache/dart-sdk/bin
+flutter config --no-analytics
+flutter doctor
+export FLUTTER_SDK=`pwd`/../flutter
+
+if [ "$IDEA_VERSION" = "4.0" -o "$IDEA_VERSION" = "4.1" ] ; then
```
Comment (devoncarew) [inline tool/github.sh:18]: What does this if statement do? Perhaps add a comment line (as bash if statements can be mysterious for people).

File: tool/github.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+# Get packages for the test data.
+(cd testData/sample_tests; echo "pub get `pwd`"; pub get)
+
+# Set up the plugin tool.
+(cd tool/plugin; echo "pub get `pwd`"; pub get)
+
+if [ "$DART_BOT" = true ] ; then
+
+  # analyze the Dart code in the repo
+  echo "pub global activate tuneup"
```
Comment (devoncarew) [inline tool/github.sh:43]: We should be able to convert this over to just `dart analyze` now (not necessarily for this PR).

File: tool/github.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  pub global activate tuneup
+  pub global run tuneup
+
+  # ensure that the edits have been applied to template files (and their target
+  # files have been regenerated)
+  ./bin/plugin generate
+
+  # show any changed file
```
Comment (devoncarew) [inline tool/github.sh:51]: nit: files

File: .github/workflows/presubmit.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
+name: presubmit
+on: [push]
+jobs:
+  windows-test:
```
Comment (devoncarew) [inline .github/workflows/presubmit.yaml:4]: Should this be named `presubmit-windows`? Is this essentially the same as the presubmit bot below?

**Review by devoncarew — APPROVED** [review]
lgtm; some questions in-line

File: tool/github.sh (hunk 1 above)
Comment (stevemessick) [reply tool/github.sh:18]: I updated the comment in the block.

File: tool/github.sh (hunk 2 above)
Comment (stevemessick) [reply tool/github.sh]: That would be more readable.

File: tool/github.sh (hunk 3 above)
Comment (stevemessick) [reply tool/github.sh]: lol, I copied that from travis.sh. It's been "file" for almost two years. I'll change it.

File: .github/workflows/presubmit.yaml (hunk 4 above)
Comment (stevemessick) [reply .github/workflows/presubmit.yaml:4]: The presubmit jobs do version-specific builds. This one runs unit tests on Windows. It is more similar to the checker job but I want to have "windows" in the name. I think this name is descriptive.

**Comment by stevemessick** [comment]
The Windows-based workflow is now disabled. Many unit tests don't run and I don't know enough about Windows to fix them.

**Pull request merged**